Improve c++ & fix some bugs

allight opened the pull request:
The settings seemed to be ignored due to not including the full name. Fixed this issue.

Made the c-like block-comments formatting a bit nicer.

allight committed:
Actually honor the holder, year and license settings.

These would always just use the defaults or undefined due to incorrect
setting strings.

Test: manual

allight committed:
Make copyright formatting for c/c++ a bit better

We want formatting to be like:

/* Copyright <year> <whoever>
 *
 * This is a copyright notice.
 *
 * It has several paragraphs.
 */

This change supports this formatting whenever the block-comment start
is '/*'.

Test: Manual (c++, html & batch).

minherz commented on src/copyrightinserter.ts:
```diff
@@ -191,15 +191,35 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.`]
+                // We want:
+                // /* the first line
+                //  * the second line
+                //  */
```
I disagree with this statement. A quick browse for comment guideline shows there are one of three approaches:
```
/*
 * LINE1
 * LINE2
 */
```
```
// LINE1
// LINE2
```
```
/* LINE1
 * LINE2 */
```

The first one is closest to your proposal. I suggest you to align to one of them. Probably first one is closest.

allight replied:
done.

minherz commented on src/copyrightinserter.ts:
```diff
@@ -191,15 +191,35 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.`]
+        var result : string = ""
+        for (const line of header.split("\n")) {
+            const new_line = cur_prefix + line
+            result += new_line.trimRight() + "\n"
```
Appending to all lines of the header including last line creates unclear behavior for a caller of the method when `post` argument is used with \newline symbol too. Either the `first_line` argument should be made `prefix` and placed in a dedicated line and then `post` is placed in the dedicated line as well or, the first line of the header should be prefixed with `first_line` and the last line of the header should be suffixed with `post` (renamed to `last_line`).

allight replied:
done

minherz commented on src/copyrightinserter.ts:
```diff
@@ -191,15 +191,35 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.`]
+                //  */
+                header = this.formatString(header, c!.blockComment[0] + " ", " * ", " " + c!.blockComment[1])
+            } else {
+                header = this.formatString(header, c!.blockComment[0] + "\n", "", c!.blockComment[1])
```
the behavior should be consistent. either opening comment should be placed in the beginning of the first line or on its own. and the same behavior should be for the closing comment. it should not depend on the specific symbols. the dedicated customization for `/**/` can be added because, as you correctly noted, it is widely accepted guidelines.

allight replied:
done

allight committed:
Tweak format and address comments.

Tweak format to have c/c++ copyright start on the line after the
initial '/*'.

Change some naming.

Test: manual

allight reviewed (commented):
done.

allight commented:
Ping.

minherz commented:
sorry for the delay with the response. i will handle it by EOW

minherz commented on src/copyrightinserter.ts:
```diff
@@ -191,15 +191,42 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.`]
+    private formatString(header: string, first_line: string, prefix: string, last_line: string): string {
+        var cur_prefix = first_line
+        var result : string = ""
+        for (const line of header.split("\n")) {
```
insert following before `for` loop to make first_line and prefix to be inserted separately:
```javascript
header = "\n" + header;
```
Then you can change [line 223](/minherz/copyright-inserter/blob/6a09fdcb1085e5bd2134dc0651bca404dc245d9a/src/copyrightinserter.ts#L223) to pass `first_line` argument as
```javascript
header = this.formatString(header, c!.blockComment[0], " * ", " " + c!.blockComment[1] + "\n")
```
See the referenced branch pr1 [line197](/minherz/copyright-inserter/blob/pr1/src/copyrightinserter.ts#L197) and [line 223](/minherz/copyright-inserter/blob/pr1/src/copyrightinserter.ts#L223)

allight replied:
Done.

allight replied:
Done.

minherz commented on src/copyrightinserter.ts:
```diff
@@ -191,15 +191,42 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.`]
     }
 
+    private formatString(header: string, first_line: string, prefix: string, last_line: string): string {
+        var cur_prefix = first_line
```
consider adding trailing semicolon (;) to the new lines. VSCode shows warning for these lines because the file type is typescript and not javascript.

allight replied:
Done.

minherz commented on src/copyrightinserter.ts:
```diff
@@ -191,15 +191,42 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.`]
+            // <Fist line of copyright>\n
+            // <etc.>\n
+            // <Block Comment End>\n
+            if (c!.blockComment[0] == "/*") {
```
use operator `===` instead of `==` to emphasis that it compares string values.

allight replied:
done

minherz commented on src/copyrightinserter.ts:
```diff
@@ -191,15 +191,42 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.`]
+            //
+            // <Block Comment Start>\n
+            // <Fist line of copyright>\n
+            // <etc.>\n
```
[OPTIONAL]
the comment should read:
```
// <Prefix><First line of copyright>\n
// <Prefix><etc.>\n

allight replied:
Done.

minherz reviewed (changes requested):
Hopefully this is the last one. I've made changes in a separate [branch](/minherz/copyright-inserter/blob/pr1/src/copyrightinserter.ts) if you would like to compare.

allight committed:
Address comments about TS style and formatting

Test: Manual.

allight committed:
Improve comment about formatting of copyright

Test: Manual

allight reviewed (commented):
Done.

minherz commented:
i will release next version of plug-in shortly

minherz commented:
published
